swagger.js

```diff
@@ -31,6 +31,15 @@ const doc = {
         },
       },
     },
+    TokenizedUser: {
+      type: 'object',
+      properties: {
+        token: { type: 'string' },
+        user: {
+          $ref: '#/definitions/PasswordlessUser',
+        },
+      },
+    },
```

Comment:
For swagger-autogen we don't need to tell it the type/properties, it will infer those. Rather it wants us to give it an example of what this object will look like and it will figure out the types from there. So for example, if I make a call to the register endpoint (via postman, or the api docs, etc) I get a response like:

```json
{
  "user": {
    "name": "John Doe",
    "userId": "64ef5885c2a489217c571253",
    "role": "user"
  }
}
```

The nice thing about swagger-autogen is we can just copy paste that exactly into our swagger.js definition here:

```suggestion
    TokenizedUser: {
      user: {
        name: "John Doe",
        userId: "64ef5885c2a489217c571253",
        role: { '@enum': ['user', 'admin'] },
      },
    },
```

Unlike the user controller endpoints, the auth endpoints don't return a full user object so we can't just use a reference to the PasswordlessUser definition above as that would make us generate misleading documentation.

Reply:
<img width="458" alt="swagger" src="https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/95896159/9585f5a2-87f3-4e5b-8381-680580388b87">